rules/fetch-data-react/rule.md

```diff
@@ -0,0 +1,145 @@
+---
+type: rule
+archivedreason: 
+title: Do you know the best libraries to fetch data in React?
+guid: cae7be1d-2313-43e1-8cc5-cfc1e77b4bb4
+uri: fetch-data-nextjs
+created: 2023-08-08T05:24:35Z
+authors:
+- title: Harry Ross
+  url: https://www.ssw.com.au/people/harry-ross
+related:
+- fetch-data-nextjs
+---
+
+While using a regular `useEffect` to run when a component is loaded to load data from an API is super easy, it may result in unnecesary duplicate requests for data. It is best to use a library for this purpose, as not only does it solve the afforementioned issue, but also comes with a variety of other useful features such as caching, background updates, and pre-fetching. 
```

Comment:
Also to mention that you have to take care and teardown the setup after the component is unloaded, because then `setData` and `setLoading` is no longer available - typically you'll see this error:

> Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.

Reply:
Made some changes - shortened some bits in this section and added errors as a reason why you would choose to use these libraries